### PR TITLE
[Fluid] Compressible NS automatic_dt considers artificial diffusion

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
**📝 Description**
Small change in compressible NS solver to enforce the use of artificial diffusivities in computation of Fourier numbers when shock capturing and automatic time step are enabled.

**🆕 Changelog**
- consider_artificial_diffusion is now overidden to the same value as `SHOCK_CAPTURING_SWITCH`.

### CODE REVIEW

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py [hunk 1]
```diff
@@ -150,18 +150,24 @@ def _create_solution_strategy(self):
             err_msg = err_msg + " - {}\n".format(key)
         raise RuntimeError(err_msg)
 
+    @classmethod
+    def _OverrideBoolParameterWithWarning(cls, parent, child, value):
+        if parent.Has(child) and parent[child].GetBool() != value:
+            KratosMultiphysics.Logger.PrintWarning("", "User-specifed {} will be overriden with {}".format(child, value))
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]:
```suggestion
        if parent.Has(child):
            if parent[child].GetBool() != value:
                KratosMultiphysics.Logger.PrintWarning("", "User-specifed {} will be overriden with {}".format(child, value))
```

Comment (ddiezrod) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]: Why? I think the original version looks better

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]: Wouldn't it crash if the `Has` is false when trying to access `parent[child]`? (here I'm assuming that all the loop clauses are evaluated, not 100% sure about how it works internally).

Comment (sunethwarna) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]: AFAIK, `and` compound statements does not execute the trailing ones if one of the leading ones are false.

```
class A:
   def Is(self, t):
      print("test")
      return t
a = A()
b = A()
if a.Is(False) and b.Is(False):
   print("D")
```

Comment (ddiezrod) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]: nope, "and" and "or" operators follow short-circuit evaluation. In the and, if first condition is false it does not access the second one. Similarly, in or, if first condition is true, second is not evaluated.

Comment (ddiezrod) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]: https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not

Comment (EduardGomezEscandell) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]: Python has short-circuiting just like C and C++.

See the [reference](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not) regarding the `and` operator:
> This is a short-circuit operator, so it only evaluates the second argument if the first one is true.

In summary, this is a safe way of accessing parameters.

Comment (ddiezrod) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]: pole

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py:156]: Very good. Thanks. I wasn't aware.

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**